### ISSUE TITLE
semantic: align TypeScript declaration conflict checks with tsc

### ISSUE THREAD

**Issue by camc314** [oxc-project/oxc#22787]
## Summary

Oxc misses several TypeScript declaration conflicts that `tsc` reports. In these cases Oxc either accepts duplicate declarations or merges symbols that TypeScript keeps invalid.

This affects semantic diagnostics and can also produce reference/declaration data for programs that TypeScript rejects.

### Duplicate simple parameters in TypeScript

```ts
function f(a, a) {}
```

TypeScript reports:

```text
TS2300: Duplicate identifier 'a'.
```

Oxc currently accepts this in TypeScript mode and creates a single parameter symbol.

### `function` + `var` duplicate in TypeScript

```ts
function f() {}
var f;
```

TypeScript reports:

```text
TS2300: Duplicate identifier 'f'.
```

Oxc currently merges the declarations into one symbol with both function and function-scoped-variable flags.

### Import/local declaration conflict in TypeScript

```ts
import { Foo } from "./m";

var Foo;
```

TypeScript reports:

```text
TS2440: Import declaration conflicts with local declaration of 'Foo'.
```

Oxc currently allows this in TypeScript mode.

## TypeScript behavior

TypeScript’s symbol exclusions keep these cases invalid:

- parameters are bound with `ParameterExcludes`
- `FunctionScopedVariableExcludes` excludes value declarations other than function-scoped variables
- import aliases are checked against conflicting local declarations

Relevant TypeScript source:

- parameter binding:
  https://github.com/microsoft/TypeScript/blob/f350b52331494b68c90ab02e2b6d0828d2a22a74/src/compiler/binder.ts#L3684-L3699
- symbol flag/exclusion definitions:
  https://github.com/microsoft/TypeScript/blob/f350b52331494b68c90ab02e2b6d0828d2a22a74/src/compiler/types.ts#L5974-L6006
- import/export declaration conflict checks:
  https://github.com/microsoft/TypeScript/blob/f350b52331494b68c90ab02e2b6d0828d2a22a74/src/compiler/checker.ts#L48396-L48454

## Oxc behavior

Oxc’s parameter binder only forces duplicate parameter errors for arrow/unique parameters, strict mode, or non-simple parameter lists:

https://github.com/oxc-project/oxc/blob/main/crates/oxc_semantic/src/binder.rs#L248-L266

Oxc’s symbol exclusions allow some TypeScript-invalid value merges:

https://github.com/oxc-project/oxc/blob/main/crates/oxc_syntax/src/symbol.rs#L81-L99

Oxc’s import binding excludes only other imports/type imports:

https://github.com/oxc-project/oxc/blob/main/crates/oxc_semantic/src/binder.rs#L335-L353

There is also an existing semantic test that currently asserts the TypeScript import/var case should be allowed:

https://github.com/oxc-project/oxc/blob/main/crates/oxc_semantic/tests/integration/symbols.rs#L458-L468

## Why this matters

These declaration conflicts are part of TypeScript’s semantic rules. Accepting or merging them makes Oxc’s diagnostics and symbol graph disagree with `tsc`, which can affect downstream tools that rely on `oxc_semantic` for TypeScript-aware binding information.

**Comment by soheilnikroo** [comment]
@camc314  I'd like to work on this. I think we can fix it by tightening TS-only binding in the binder — forbid duplicate params in TS, use stricter `var` excludes so it doesn't merge with `function`, and handle import + `var` conflicts (plus fix that test). What's your opinion on that flow?
